docs/docs/guides/integrations/mcp.md

```diff
@@ -0,0 +1,236 @@
+
+You will find a trace timeline as shown below:
+
+[![mcp_trace_timeline.png](imgs/mcp/mcp_trace_timeline.png)](https://wandb.ai/ayut/mcp_example/weave/traces?filter=%7B%22opVersionRefs%22%3A%5B%22weave%3A%2F%2F%2Fayut%2Fmcp_example%2Fop%2Frun_client%3A*%22%5D%7D&peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bbe-cc5e-7012-b45f-bf10617d8c1e%3FhideTraceTree%3D0)
+
+By default, you will just see the `run_client` traces. Click on the Ops selection box and select "All Calls"
+
+[!mcp_all_calls.png](imgs/mcp/mcp_all_calls.png)]
+
+Doing so will show you `FastMCP` methods (tools, resources, prompts) traced by the integration. You can see the arguments given to the tools and returned values.
+
+[!mcp_fastmcp.png](imgs/mcp/mcp_fastmcp.png)](https://wandb.ai/ayut/mcp_example/weave/traces?peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bc2-aca1-7021-a626-aecfe677b1b4%3FhideTraceTree%3D0)
+
+### Understanding the Example
+
+#### Server-Side Code
+
+The example server (`example_server.py`) defines:
+
+1. **Tools**: Functions like `add()`, `calculate_bmi()`, and `fetch_weather()`
+2. **Resources**: Data endpoints like `greeting://{name}`, `config://app`, and `users://{user_id}/profile`
+3. **Prompts**: Templates like `review_code()` and `debug_error()`
+
+All of these components are automatically traced by Weave, allowing you to monitor their usage, inputs, and outputs.
+
+#### Client-Side Code
+
+The example client (`example_client.py`) demonstrates:
+
+1. How to connect to an MCP server
+2. How to discover available tools, resources, and prompts
+3. How to call tools with parameters
+4. How to access resources with URIs
+5. How to generate prompts with arguments
+
+All client-side operations are also traced by Weave, creating a complete picture of the client-server interaction.
+
+## Using the Integration
+
+### Client-Side Integration
+
+To enable tracing on an MCP client:
+
+```python
+import weave
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+
+# Initialize Weave
+weave_client = weave.init("my-project")
+
+# The MCP client operations will be automatically traced
+async with stdio_client(server_params) as (read, write):
+    async with ClientSession(read, write) as session:
+        # Initialize the connection
+        await session.initialize()
+        
+        # Call a tool (this will be traced)
+        result = await session.call_tool("add", arguments={"a": 1, "b": 2})
+        
+        # Read a resource (this will be traced)
+        resource = await session.read_resource("greeting://user")
+        
+        # Get a prompt (this will be traced)
+        prompt = await session.get_prompt("review_code", arguments={"code": "print('Hello')"})
+```
+
+### Server-Side Integration
+
+To enable tracing on an MCP server:
+
+```python
+import weave
+from mcp.server.fastmcp import FastMCP
+
+# Initialize Weave
+weave_client = weave.init("my-project")
```

Comment:
use the code highlighting call and say that its all you need

add `# highlight-next-line` to the line above what you want to highlight